feat(kselect): add clearable prop (#749)
Kong/kongponents: docs/components/select.md

<KSelect :items="items" />
```

> The `select` style displays the selected item in the textbox and also displays a chevron. To deselect the item, you need to

Reviewer: ```suggestion
The `select` style displays the selected item in the text box and also displays a chevron. To allow deselecting the item, you need to
```

<KSelect appearance='button' :items="items" />
```

### clearable

> The `clearable` prop is used to enable deselecting the selected item when `appearance` is `'select'`.

Reviewer: ```suggestion
The `clearable` prop is used to enable deselecting the selected item when `appearance` is `'select'`. Defaults to `false`.
```

<div>

>   <KSelect appearance='select' :items="deepClone(defaultItems)" clearable />

Reviewer: ```suggestion
  <KSelect appearance="select" :items="deepClone(defaultItems)" clearable />
```

</div>

```html

> <KSelect appearance='select' :items="items" clearable />

Reviewer: ```suggestion
<KSelect appearance="select" :items="items" clearable />
```
